### PR TITLE
Clear some properties in `ParticleBuffer.destroy()`.

### PR DESCRIPTION
**Pull request by finscn**
* Clear some properties in `ParticleBuffer.destroy()`.
* improve the shader.

### CODE REVIEW

File: src/particles/webgl/ParticleShader.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ export default class ParticleShader extends Shader
                 'varying vec4 vColor;',
 
                 'void main(void){',
-                '   vec2 v = aVertexPosition;',
+                '   vec2 v = vec2(0.0, 0.0);',
```
Comment (bigtimebuddy) [inline src/particles/webgl/ParticleShader.js]: Could you explain this change @finscn?

Comment (finscn) [reply src/particles/webgl/ParticleShader.js]: The results of  `vec2 v = aVertexPosition;` & `vec2 v = vec2(0.0, 0.0);` are both OK.

But in the semantic  , the ` vec2 v = aVertexPosition;` is not accuracy  , and needless.

In this shader , the  `v` is a new vec2 , we don't need  copy or clone `aVertexPosition` to it. 

Use `v = aVertexPosition` will puzzle users (the people that read the code).

Comment (bigtimebuddy) [reply src/particles/webgl/ParticleShader.js]: Thanks for the explanation.

Comment (englercj) [reply src/particles/webgl/ParticleShader.js]: I prefer to just initialize it when I do shaders:

```glsl
 vec2 v(
    (aVertexPosition.x) * cos(aRotation) - (aVertexPosition.y) * sin(aRotation),
    (aVertexPosition.x) * sin(aRotation) + (aVertexPosition.y) * cos(aRotation)
);
```

**Pull request merged**